internal/setup/setup.go

```diff
@@ -562,14 +598,21 @@ func parsePackage(baseDir, pkgName, pkgPath string, data []byte) (*Package, erro
 			var mutable bool
 			var until PathUntil
 			var arch []string
+			var generate GenerateKind
 			if strings.ContainsAny(contPath, "*?") {
 				if yamlPath != nil {
-					if !yamlPath.SameContent(&zeroPath) {
+					if len(yamlPath.Generate) == 0 {
+						kinds = append(kinds, GlobPath)
+					}
+					tempPath := zeroPath
+					tempPath.Generate = yamlPath.Generate
+					if !yamlPath.SameContent(&tempPath) {
```

Comment:
This is nice and compact integrating `generate` into the rest of the logic. However, that comes at the cost of having less specific error messages. For example when using `generate` and `text` do we want the message to be `slice %s_%s path %s has invalid wildcard options` or do we want it to be something specific to generate.

Anyhow, `tempPath` -> `zeroPathGenerate` I think is clearer.

Reply:
I have another version where the check for glob is a little bit more independent, tell me what you think:
```go
if yamlPath != nil && yamlPath.Generate != "" {
	if yamlPath.Generate != "chisel-state" {
		return nil, fmt.Errorf("slice %s_%s has invalid 'generate' for path %s: %q",
			pkgName, sliceName, contPath, yamlPath.Generate)
	}
	zeroPathGenerate := zeroPath
	zeroPathGenerate.Generate = yamlPath.Generate
	if !yamlPath.SameContent(&zeroPathGenerate) {
		return nil, fmt.Errorf("slice %s_%s path %s has invalid generate options",
			pkgName, sliceName, contPath)
	}
	// This block can be here or below were you had placed it.
	if !strings.HasSuffix(contPath, "/**") {
		return nil, fmt.Errorf("slice %s_%s path %s must end with '/**' for 'generate' to be valid", pkgName, sliceName, contPath)
	}
	if strings.ContainsAny(contPath[:len(contPath)-2], "*?") {
		return nil, fmt.Errorf("path %s must not contain any other wildcard characters except trailing ** for 'generate: %s' to be valid",
			contPath, "")
	}
	generate = yamlPath.Generate
	// end of block.
} else if strings.ContainsAny(contPath, "*?") {
	if yamlPath != nil {
		if !yamlPath.SameContent(&zeroPath) {
			return nil, fmt.Errorf("slice %s_%s path %s has invalid wildcard options",
				pkgName, sliceName, contPath)
		}
	}
	kinds = append(kinds, GlobPath)
}

if yamlPath != nil {
	...


```

Reply:
Good point.I like your second suggestion. Added in 272d559. 
Please note the changes as this enforces no other options to be used with "generate".